libs/core/src/lib/utils/directives/truncation/truncation.directive.ts

```diff
@@ -0,0 +1,55 @@
+import { AfterViewInit, Directive, ElementRef, Input } from '@angular/core';
+
+@Directive({
+    // eslint-disable-next-line @angular-eslint/directive-selector
+    selector: '[fd-truncate]'
+})
+export class TruncationDirective implements AfterViewInit {
+    @Input() charLength: number;
+    @Input() pixLength: number;
+
+    /** @hidden */
+    private defaultTruncationByChar = 30;
+
+    constructor(private _elementRef: ElementRef) {}
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        const rootElement = this._elementRef.nativeElement;
+        rootElement.title = rootElement.textContent; // adds title to element to enable mouse over tooltip
+
+        if (this.pixLength) {
+            this._truncateByPixel(rootElement);
+        } else {
+            this._truncateByChar(rootElement);
+        }
+    }
+
+    /** @hidden
+     * Truncates element by width
+     */
+    private _truncateByPixel(_element): void {
+        _element.style.overflow = 'hidden';
+        _element.style.textOverflow = 'ellipsis';
+        _element.style.whiteSpace = 'nowrap';
+        _element.style.width = `${this.pixLength}px`;
+    }
+
+    /** @hidden
+     * Truncates element's text by characters
+     */
+    private _truncateByChar(_element): void {
+        console.log(this._elementRef);
```

Comment:
I know this is a WIP so maybe you were already planning on removing it, but we try to avoid console logs in the codebase